executor/aggfuncs/func_group_concat.go

```diff
@@ -389,21 +413,25 @@ func (e *groupConcatOrder) UpdatePartialResult(sctx sessionctx.Context, rowsInGr
 		for _, byItem := range e.byItems {
 			d, err := byItem.Expr.Eval(row)
 			if err != nil {
-				return 0, err
+				return memDelta, err
 			}
+			memDelta += GetDatumMemSize(&d)
 			sortRow.byItems = append(sortRow.byItems, d.Clone())
 		}
-		truncated := p.topN.tryToAdd(sortRow)
+		oldMem := p.topN.currSize
+		truncated, truncatedDatumMemSize := p.topN.tryToAdd(sortRow)
+		newMem := p.topN.currSize
+		memDelta += int64(newMem-oldMem) - truncatedDatumMemSize
```

Comment:
could we move L421 and L423 into `tryToAdd`?

Reply:
ok